# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/sakshi987123/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

In [9]:
!git clone https://github.com/sakshi987123/flyrank-ml-internship.git

Cloning into 'flyrank-ml-internship'...
remote: Enumerating objects: 120, done.
remote: Counting objects: 100% (120/120), done.
remote: Compressing objects: 100% (77/77), done.
remote: Total 120 (delta 36), reused 94 (delta 27), pack-reused 0 (from 0)
Receiving objects: 100% (120/120), 1.84 MiB | 10.67 MiB/s, done.
Resolving deltas: 100% (36/36), done.


In [10]:
%cd flyrank-ml-internship

/content/flyrank-ml-internship


In [11]:
import os
print(os.listdir())

['notebooks', 'work', 'AGENTS.md', 'skills', 'DATA_USE.md', 'scripts', '.github', 'LICENSE', '.gitignore', 'outputs', 'CLAUDE.md', 'data', 'README.md', 'submission', '.git', 'GUIDE.md', 'SETUP.md', 'requirements.txt', 'docs']


## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

## My Lane

I chose the **Ranking Signal Analysis** lane.

In the starter notebook, I observed that search volume alone had almost no correlation with impressions, suggesting that ranking performance depends on multiple factors rather than a single metric. I want to analyze which page and search signals are most strongly associated with better search rankings and visibility. This analysis can help prioritize SEO improvements based on evidence instead of assumptions.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

## Research Question

**Search Question:**
Which ranking signals are most strongly associated with better search performance?

**Decision:**
Which pages should be prioritized for SEO improvements based on their ranking signals?

**Unit of Analysis:**
Individual web pages.

**Output:**
A ranked list of important ranking signals and pages that could benefit from optimization.

**Action:**
The SEO or content team can improve page titles, content quality, metadata, or technical SEO based on the identified signals.

**Cost of a Wrong Recommendation:**
If the wrong ranking signals are prioritized, valuable time and effort may be spent on changes that do not improve search performance, while better optimization opportunities are overlooked.

**Why Data or ML Can Help:**
The dataset contains multiple search-related features such as impressions, CTR, average position, search volume, engagement metrics, and content information. Data analysis and machine learning can identify patterns among these features that are difficult to discover manually.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


In [12]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Rows:", len(df))
print("Columns:", len(df.columns))
print("Correlation:", round(df["search_volume"].corr(df["impressions_90d"]), 3))
print("Average CTR:", round(df["ctr"].mean(), 3))
print("Average Position:", round(df["avg_position"].mean(), 2))

Rows: 30000
Columns: 44
Correlation: 0.001
Average CTR: 0.511
Average Position: 16.34


## Quick Look at the Data

The starter dataset contains **30,000 web pages** and **44 features**, providing enough information to analyze search performance.

The correlation between **search volume** and **impressions** is only **0.001**, showing that search volume alone is not a strong indicator of page visibility.

The average **CTR is 0.511** and the average **search position is 16.34**, indicating that multiple ranking-related signals should be analyzed together rather than relying on a single metric.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

## Careful Words

This project identifies observed relationships between ranking signals and search performance. The results provide decision support and do not prove causation or explain Google's ranking algorithm. Any recommendations should be validated before implementation.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.